src/databasetasks.cpp

```diff
@@ -85,6 +85,13 @@ void DatabaseTasks::runTask(const DatabaseTask& task)
 
 void DatabaseTasks::flush()
 {
+	taskLock.lock();
+	flushInternal();
+	taskLock.unlock();
+}
+
+void DatabaseTasks::flushInternal()
+{
 	while (!tasks.empty()) {
 		runTask(tasks.front());
```

Comment:
Btw if the lock inside the flush function can deadlock then the DatabaseTasks::shutdown() may make a deadlock as both functions(DatabaseTasks::flush() and DatabaseTasks::shutdown()) are called from the dispatcher thread. Also the query callback will be only called in the next dispatcher cycle(g_dispatcher.addTask(createTask(std::bind(task.callback, result, success)));) so how it will cause a deadlock?